samples/extensions/descriptor_indexing/descriptor_indexing.cpp

```diff
@@ -31,6 +31,26 @@ DescriptorIndexing::DescriptorIndexing()
 	// Works around a validation layer bug with descriptor pool allocation with VARIABLE_COUNT.
 	// See: https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/2350.
 	add_device_extension(VK_KHR_MAINTENANCE1_EXTENSION_NAME);
+
+#if defined(PLATFORM__MACOS) && TARGET_OS_OSX
+	// On macOS use layer setting to configure MoltenVK for using Metal argument buffers (needed for descriptor indexing)
+	add_instance_extension(VK_EXT_LAYER_SETTINGS_EXTENSION_NAME, /*optional*/ true);
+
+	VkLayerSettingEXT layerSetting;
+	layerSetting.pLayerName   = "MoltenVK";
+	layerSetting.pSettingName = "MVK_CONFIG_USE_METAL_ARGUMENT_BUFFERS";
+	layerSetting.type         = VK_LAYER_SETTING_TYPE_INT32_EXT;
+	layerSetting.valueCount   = 1;
+
+	static const int32_t useMetalArgumentBuffers = 1;
```

Comment:
The layer settings API wants a pointer to the setting and not its value (i.e. not a copy), in this case:

`layerSetting.pValues = &useMetalArgumentBuffers`

Given the vector of layer settings is used outside of the sample's constructor context (i.e. after the constructor's completion), you need to keep the variable around so it has not been deallocated when referenced by `vkCreateInstance()`.  This would be made easier in ref counting implementations, but not the case here.

Reply:
I mean... a short comment in the source ;)

Reply:
Sorry :)

I will do this when I get home tonight (EDT time).